source/common/common/thread.h

```diff
@@ -43,13 +43,12 @@ typedef std::unique_ptr<Thread> ThreadPtr;
  */
 class MutexBasicLockable : public BasicLockable {
 public:
-  void lock() override { mutex_.lock(); }
-  bool try_lock() override { return mutex_.try_lock(); }
-  void unlock() override { mutex_.unlock(); }
+  void lock() EXCLUSIVE_LOCK_FUNCTION() override { mutex_.Lock(); }
```

Comment:
Should these annotations also go on the interface/base-class?

Reply:
They are also on BasicLockable.  But they need to go here as well or the compiler failures with "mutex_ held at end of function".